### PR TITLE
Update spec draft, explainer, examples for EditContext

### PR DESCRIPTION
**Pull request by alexkeng**
This PR adds Editor's draft for EditContext spec (index.html), revises the Explainer (explainer.md), and adds new example (canvas_editContext.html).

### CODE REVIEW

**Comment by travisleithead** [comment]
Easy readable version for review here: https://alexkeng.github.io/editing/docs/EditContext/index.html

File: docs/EditContext/index.html [hunk 1]
```diff
@@ -0,0 +1,914 @@
...
+
+    class EditingView {
+        render() {
+            // render the text (implementation omitted for brevity)
+
+            // render the selection (implementation omitted for brevity)
+
+            // render the IME decoration
+            this.model.textFormats.forEach( textFormat => {
+                let lineStartX = anchorX + textFormat.rangeStart * charWidth;
```
Comment (travisleithead) [inline docs/EditContext/index.html:229]: A few of these variables are used, but not defined. Might be good to note what they are in comments? (I understand that this is example code and that these variables are largely self-explanatory...)

Comment (travisleithead) [reply docs/EditContext/index.html:229]: e.g., `anchorX`, `charWidth`, `thickWidth`, `lineY`...

File: docs/EditContext/index.html [hunk 2]
```diff
@@ -0,0 +1,914 @@
...
+    editContext.addEventListener("textupdate", e => {
+        editingController.handleTextUpdate(e.updateRangeStart, e.updateRangeEnd, e.updatetext,
+                                        e.newSelectionStart, e.newSelectionEnd,
+                                        e.compositionStart, e.compositionEnd)
+    });
+
+    // EditContext will also receive textformatupdate event for IME decoration.
+    // Ex. thin/thick underline for the "phrase mode" in Japanese IME.
+    editContext.addEventListener("textformatupdate", e => {
+        editingcontroller.updateTextFormatsUpdate(e.getTextFormats());
```
Comment (travisleithead) [inline docs/EditContext/index.html]:
```suggestion
        editingcontroller.handleTextFormatUpdate(e.getTextFormats());
```

File: docs/EditContext/index.html [hunk 3]
```diff
@@ -0,0 +1,914 @@
...
+        }
+    }
+
+    class EditingController {
+        handleSelectionChange() {
+            // Authors are responsible to map the selection from the DOM space to the
+            // plain text space. One approach is to use range.toString() to get the plain text
+            // before the selection start/end, and use the text length as the selection index.
+            let s = document.getSelection()
+            let range = document.createRange()
```
Comment (travisleithead) [inline docs/EditContext/index.html]:
```suggestion
            let range = new Range()
```

Comment (travisleithead) [reply docs/EditContext/index.html]: (Just a more modern approach...)

**Review by travisleithead — CHANGES_REQUESTED** [review]
Just a few minor changes. Looks excellent. Great update!

File: docs/EditContext/index.html (hunk 3 above)
Comment (alexkeng) [reply docs/EditContext/index.html]: sure, updated to new Range()

File: docs/EditContext/index.html (hunk 2 above)
Comment (alexkeng) [reply docs/EditContext/index.html]: good catch!

File: docs/EditContext/index.html (hunk 1 above)
Comment (alexkeng) [reply docs/EditContext/index.html:229]: sure! comment added

File: docs/EditContext/index.html (hunk 3 above)
Comment (alexkeng) [reply docs/EditContext/index.html]: thanks for the quick review!

**Comment by xfq** [comment]
FWIW, the link to the demo video in [Example Application](https://github.com/w3c/editing/blob/23711b55bb7a8b388d8691372b79441152107ebe/docs/EditContext/explainer.md#example-application) emitted:

> Sorry, the link has expired.
> The link was set to expire after a certain amount of time. Please contact the person who shared this link with you.

I don't know if there's an issue with the URL or it's just me.

**Review by travisleithead — APPROVED** [review]
Awesome. Lets get this merged.

**Pull request merged**